### PR TITLE
Add new zpm and uv battery

### PR DESCRIPTION
**Pull request by SLiMylya**
add new battery

change recipe

### CODE REVIEW

**Comment by draknyte1** [comment]
We're not changing the storage of the ultimate battery.

**Comment by Techlone** [comment]
@draknyte1 what's a problem with it?

**Comment by Blood-Asp** [comment]
Personally i would prefer to keep MAX Voltage only as transfer Voltage, not for storage.

**Comment by SapientMC** [comment]
@Blood-Asp we can change MAX voltage to 2M eU/t

**Comment by Blood-Asp** [comment]
I mean the highest Power tier should be Transfer only. Same as EV for IC2 alone.

**Comment by SapientMC** [comment]
@Blood-Asp why? This will simply require a transformer at both ends of the line.

**Comment by draknyte1** [comment]
It should stay as a max value. 
Why change it? I disapprove of this PR until the original values for max are restored.

**Review by draknyte1 — CHANGES_REQUESTED** [review]
Restore Max voltage battery capacity.

**Comment by Antillar** [comment]
What about adding the most expensive battery craft (currently requested Max) to be UV, but already charged? It would set new goals for players in the endgame. 
The recipe can be made more expensive then.

**Comment by SLiMylya** [comment]
change

or delete her? this battery from achievement. Ultimate battery for the game, because the battery buffer with it can be automated

**Comment by draknyte1** [comment]
Leave the old battery as is..

If you want a zpm battery, add a config option to make empty ZPM's craftable and rechargeable.

**Comment by SLiMylya** [comment]
old battery on max capacity not  automated

**Comment by draknyte1** [comment]
I don't understand sorry. 
MAX violate will, as blood suggests, be for power transfer and not storage.

**Comment by SLiMylya** [comment]
Sorry, max capacity

File: src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java [hunk 1]
```diff
@@ -438,17 +438,16 @@ public GT_MetaGenerated_Item_01() {
...
 
-        ItemList.Energy_LapotronicOrb2.set(addItem(tLastID = 599, "Lapotronic Energy Orb Cluster", "", new Object[]{new TC_Aspects.TC_AspectStack(TC_Aspects.ELECTRUM, 16L), new TC_Aspects.TC_AspectStack(TC_Aspects.VITREUS, 16L), new TC_Aspects.TC_AspectStack(TC_Aspects.POTENTIA, 16L), OrePrefixes.battery.get(Materials.Ultimate)}));
+        ItemList.Energy_LapotronicOrb2.set(addItem(tLastID = 599, "Lapotronic Energy Orb Cluster", "Reusable battery", new Object[]{new TC_Aspects.TC_AspectStack(TC_Aspects.ELECTRUM, 16L), new TC_Aspects.TC_AspectStack(TC_Aspects.VITREUS, 16L), new TC_Aspects.TC_AspectStack(TC_Aspects.POTENTIA, 16L), OrePrefixes.battery.get(Materials.Ultimate)}));
         setElectricStats(32000 + tLastID, 1000000000L, GT_Values.V[6], 6L, -3L, true);
 
-        ItemList.ZPM2.set(addItem(tLastID = 605, "Ultimate Battery", "Fill this to win minecraft", new Object[]{new TC_Aspects.TC_AspectStack(TC_Aspects.ELECTRUM, 64L), new TC_Aspects.TC_AspectStack(TC_Aspects.VITREUS, 16L), new TC_Aspects.TC_AspectStack(TC_Aspects.POTENTIA, 64L)}));
-        setElectricStats(32000 + tLastID, Long.MAX_VALUE, GT_Values.V[8], 8L, -3L, true);
```
Comment (Dragon2488) [inline src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java:451]: Do not touch Ultimate Battery max charge. Or make this change optional.

**Review by Dragon2488 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java (hunk 1 above)
Comment (draknyte1) [reply src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java:451]: Easier to just not change it and make a UV tier battery that holds less. 

Then make a config option to make  zpm craftable and rechargeable

**Comment by SLiMylya** [comment]
Create zpm, create UV capacity 100,000,000,000.and do not touch ultimate battery. I understand correctly?

**Comment by draknyte1** [comment]
ZPM battery exists, it is single use. 
Just make config option to add recipe and allow it to be rechargeable.

**Comment by SLiMylya** [comment]
updated pr

**Comment by SLiMylya** [comment]
added config EnableZPMandUVBatteries default false

**Comment by Blood-Asp** [comment]
From what i can see are the requested changes done. So re-rewiew?

**Pull request merged**